### ISSUE TITLE
SecureView. Download and Copy buttons are available for secure view file in the search result

### ISSUE THREAD

**Issue by ScharfViktor** [owncloud/web#11189]
ocis | 6.1.0+0b8559f91
web 9.1.0

started using https://github.com/owncloud/web/blob/master/docker-compose.yml

Steps: 
- `admin` upload `.png` file -> (strange -no preview in the file list)
- `admin` add tag `secure` to `.png` file
- `admin` shares `.png` to `einstein` with secure view role
- `einstein` tries to find `.png` file by name and goes to search result

Actual: 
1. `einstein` cannot see/add tags in the file in Shares with me page but can see tag `secure` in the filter in the search result
2. buttons `Download` and `Copy` are available for `.png` file
3. `einstein` can see file preview

<img width="1107" alt="Screenshot 2024-07-15 at 12 04 58" src="https://github.com/user-attachments/assets/1d140e84-daf1-4497-9060-ba22855776a8">

https://github.com/user-attachments/assets/afae6d2c-7581-48b4-b5bc-c65beebd8cfc

**Comment by AlexAndBear** [comment]
I don't think with the current implementation of the search service this is not preventable, as we don't get the share permissions

**Comment by ScharfViktor** [comment]
then we need add permission because secure view feature becomes insecure:

case: 
- admin creates `.txt` file with secure view
- recipient can only open this file in collabora if -> he is in Shares with me page 
- recipient goes to the search result -> he can open file using Onlyoffice


https://github.com/user-attachments/assets/465a8972-ceed-41ed-b765-61ec78612ade

**Comment by JammingBen** [comment]
Can't we check for the missing DAV permission?

**Comment by ScharfViktor** [comment]
we added `X: Securely Viewable` https://owncloud.dev/apis/http/webdav/#supported-webdav-properties but response doesn't contains it. I'll create issue in ocis repo

<img width="1217" alt="Screenshot 2024-07-15 at 16 06 59" src="https://github.com/user-attachments/assets/cec6aaac-63c8-419a-8283-c0e0248d70cc">

**Comment by AlexAndBear** [comment]
How this feature is called secure at all if the file can be opened ? 🫨 
no matter if the permissions coming back from the server correct or incorrect ?

**Comment by micbar** [comment]
> How this feature is called secure at all if the file can be opened with something else than collabora ? 🫨
no matter if the permissions coming back from the server correct or incorrect 

Normal downloads are prohibited.

We made it possible to open the File in the webOffice.

Currently it "works as designed" except the full text search.

@butonic we didn't consider that "onlyoffice" is active on the same instance. Can we limit that?

**Comment by micbar** [comment]
Making this P1 because this should trigger a new rolling release.

**Comment by AlexAndBear** [comment]
> Making this P1 because this should trigger a new rolling release.

I don't think we need any code changes in web as soon the permission will be exposed via search API, everything should work out of the box. 
At least I designed it that way  🤔

**Comment by micbar** [comment]
Transfer to ocis
